snoop/data/management/commands/filestats.py

```diff
@@ -0,0 +1,181 @@
+"""Command to get statistics for filetypes that exist in collections."""
+
+from django.core.management.base import BaseCommand
+from ... import models
+from ... import collections
+from django.db.models import Count
+from django.db.models import Sum
+from django.db import connections
+from django.db.models.expressions import RawSQL
+
+from ...analyzers import archives
+from ...analyzers import tika
+from ...analyzers import email
+from ...analyzers import exif
+from ...analyzers import html
+from ... import filesystem
+
+
+SUPPORTED_MIME_TYPES = (archives.ARCHIVES_MIME_TYPES
+                        .union(tika.TIKA_MIME_TYPES)
+                        .union(filesystem.EMLX_EMAIL_MIME_TYPES)
+                        .union(email.OUTLOOK_POSSIBLE_MIME_TYPES)
+                        .union(filesystem.RFC822_EMAIL_MIME_TYPES)
+                        .union(exif.EXIFREAD_MIME_TYPES)
+                        .union(html.HTML_MIME_TYPES))
+
+
+def truncate_size(size):
+    """Generate a truncated number for a given number.
+
+    This is needed to anonymize the statistics, so they can't be traced back
+    to some dataset.
+    """
+    return round(size, -((len(str(size))) - 1))
+
+
+def get_top_mime_types(collections_list, print_supported=True):
+    """Return a dictionary of mime-types that occupy most space in collections.
+
+    Args:
+        collections_list: A list of collections that will be analyzed.
+        print_supported: When False only analyzes unsupported filetypes.
+    """
+    res = {}
+    for col in collections_list:
+        collection = collections.ALL[col]
+        with collection.set_current():
+            queryset_mime = models.Blob.objects.all().values('mime_type', 'magic')\
+                .annotate(total=Count('mime_type')).annotate(size=Sum('size'))\
+                .order_by('-size')
+            if not print_supported:
+                queryset_mime = queryset_mime.exclude(mime_type__in=SUPPORTED_MIME_TYPES)
+            for mtype in queryset_mime:
```

Comment:
Stop after the top `--row-count` / 100 mime types for each request:

```python3
queryset_mime = queryset_mime[:row_count]
```

You should also rearrange the lines above so you do the `.exclude` first, then the `values`, `annotate`, then `order_by`, and last is the limit `[:row_count]`. Django may complain because you're doing `.exclude` after `.order_by`; check docs.